crates/wash-cli/src/plugin.rs

```diff
@@ -0,0 +1,328 @@
+    Uninstall(PluginUninstallCommand),
+    /// List installed plugins
+    #[clap(name = "list", alias = "ls")]
+    List(PluginListCommand),
+}
+
+#[derive(Parser, Debug, Clone)]
+pub struct PluginCommonOpts {
+    /// Path to plugin directory. Defaults to $HOME/.wash/plugins.
+    #[clap(long = "plugin-dir", env = "WASH_PLUGIN_DIR")]
+    pub plugin_dir: Option<PathBuf>,
+}
+
+#[derive(Debug, Clone, Parser)]
+pub struct PluginInstallCommand {
+    #[clap(flatten)]
+    pub oci_auth: AuthOpts,
+
+    /// URL of the plugin to install. Can be a file://, http://, https://, or oci:// URL.
+    #[clap(name = "url")]
+    pub url: String,
+
+    /// Digest to verify plugin against. For OCI manifests, this is the digest format used in the
+    /// manifest. For other types of plugins, this is the SHA256 digest of the plugin binary.
+    #[clap(short = 'd', long = "digest")]
+    pub digest: Option<String>,
+
+    /// Allow latest artifact tags (if pulling from OCI registry)
+    #[clap(long = "allow-latest")]
+    pub allow_latest: bool,
+
+    /// Whether or not to update the plugin if it is already installed. Defaults to false
+    #[clap(long = "update")]
+    pub update: bool,
+
+    #[clap(flatten)]
+    pub opts: PluginCommonOpts,
+}
+
+#[derive(Debug, Clone, Parser)]
+pub struct PluginUninstallCommand {
+    /// ID of the plugin to uninstall
+    #[clap(name = "id")]
+    pub plugin: String,
+
+    #[clap(flatten)]
+    pub opts: PluginCommonOpts,
+}
+
+#[derive(Debug, Clone, Parser)]
+pub struct PluginListCommand {
+    #[clap(flatten)]
+    pub opts: PluginCommonOpts,
+}
+
+pub async fn handle_command(
+    cmd: PluginCommand,
+    output_kind: OutputKind,
+) -> anyhow::Result<CommandOutput> {
+    match cmd {
+        PluginCommand::Install(cmd) => handle_install(cmd, output_kind).await,
+        PluginCommand::Uninstall(cmd) => handle_uninstall(cmd, output_kind).await,
+        PluginCommand::List(cmd) => handle_list(cmd, output_kind).await,
+    }
+}
+
+pub async fn handle_install(
+    cmd: PluginInstallCommand,
+    output_kind: OutputKind,
+) -> anyhow::Result<CommandOutput> {
+    let plugin_dir = ensure_plugin_dir(cmd.opts.plugin_dir).await?;
+    let spinner = Spinner::new(&output_kind)?;
+    // Write the data to a temp file that will be cleaned and then we can move it to its real
+    // location if everything is successful.
+    let tempdir =
+        tempfile::tempdir().context("Unable to create temp directory for plugin download")?;
+    let temp_location = tempdir.path().join("temp_plugin.wasm");
+    let mut file = tokio::fs::OpenOptions::new()
+        .create(true)
+        .truncate(true)
+        .write(true)
+        .read(true)
+        .open(&temp_location)
+        .await
+        .context("Unable to create temp file for plugin download")?;
+
+    let (scheme, rest) = cmd
+        .url
+        .split_once("://")
+        .context("Invalid URL. It should contain a scheme (e.g. file://)")?;
```

Comment:
nit: does this mean `wash plugin install ./myplugin.wasm` is invalid and should be `wash plugin install file://./myplugin.wasm`? Could we just have the fallback here be an `unwrap_or_else(|_| (None, cmd.url))` and allow no schemes to be files?

As I'm writing this, I realize the difficulty in figuring out oci/http/file from just a plain string, so it makes sense. The help text addresses this well IMO so I'm just going to resolve this